src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -682,7 +682,25 @@ public GHCompare getCompare(GHCommit id1, GHCommit id2) throws IOException {
     }
 
     public GHCompare getCompare(GHBranch id1, GHBranch id2) throws IOException {
-        return getCompare(id1.getName(),id2.getName());
+
+        String fqid1 = id1.getName();
+        String fqid2 = id2.getName();
+
+        try {
+            // try to grab the org name and qualify the branch name
+            String owner1 = id1.getOwner().getOwnerName();
+            String owner2 = id2.getOwner().getOwnerName();
+            if (!owner1.equals(owner2)) {
+                fqid1 = String.format("%s:%s",owner1,id1.getName());
+                fqid2 = String.format("%s:%s",owner2,id2.getName());
+            }
+        }
+        catch (RuntimeException e) {
```

Comment:
bad practice to catch everything and ignore. Should catch only when we request owner.